actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -2468,9 +2468,19 @@ def submit(value = nil, options = {})
       #   #      <strong>Ask me!</strong>
       #   #    </button>
       #
+      #   button do |text|
+      #     content_tag(:strong, text)
+      #   end
+      #   # => <button name='button' type='submit'>
+      #   #      <strong>Create post</strong>
+      #   #    </button>
+      #
       def button(value = nil, options = {}, &block)
         value, options = nil, value if value.is_a?(Hash)
         value ||= submit_default_value
+        if block_given?
+          block = Proc.new { yield(value) }
```

Comment:
I think this is the new `value` so we can do:

```suggestion
          value = yield(value)
```

Don't that have the same behavior?

Reply:
That's viable, but if we re-assign `value`, would we also need to either _skip_ passing the block, or set `block = nil` so that it isn't yielded twice?

Reply:
Or change the return expression to totally omit the block, always:

```ruby
if block_given?
  value = yield(value)
end

@template.button_tag(value, options)
```

Reply:
@rafaelfranca I've done this in https://github.com/rails/rails/pull/38550/commits/a053f1e19a0d5e7a856408f68707b65b9b3a1860, I hope the rest of the suite passes. 🤞